fastats/linear_algebra/ols.py

```diff
@@ -204,6 +204,33 @@ def t_statistic(A, b):
     return betas / se
 
 
+def drop_missing(A, b):
+    """
+    Returns a filtration of A (features) and b (targets) where all
+    values are not NaN, with the intention that OLS can then be
+    performed on dense / 'complete' observations.
+
+    This is analogous to the statsmodels missing='drop' mechanism.
+    """
+    m = A.shape[0]
+    assert m == len(b)
+
+    A_bar = np.empty_like(A)
+    b_bar = np.empty_like(b)
+
+    ctr = 0
+
+    for i in range(m):
+        feature_row = A[i, :]
+        target = b[i]
+        if np.all(~np.isnan(feature_row)) and ~np.isnan(target):
```

Comment:
Wouldn't it be better to call `isnan` only once and use boolean masks instead?

Reply:
Do you mean by concatenating the feature matrix and target vector, then calling isnan?  I wired it up like this initially, but the didn't seem to read as well (to me) - i.e. for an observation to be valid, all feature values and the corresponding target must not be NaN.  But I'm easy (and if you meant something else, let me know)

Reply:
Yes, that's what I meant - two `np.isnan` arrays for `A` and `b`. I have vague memory of that being a bit faster than calling it in a loop - but if it's not, this is definitely more readable!

Reply:
Cool, I will have a look when I next get a mo - thanks for the suggestion